rapid_pro_tools/rapid_pro_client.py

```diff
@@ -0,0 +1,120 @@
+import datetime
+
+from core_data_modules.traced_data import TracedData, Metadata
+from core_data_modules.util import TimeUtils
+from temba_client.v2 import TembaClient
+
+
+class RapidProClient(object):
+    def __init__(self, server, token):
+        self.rapid_pro = TembaClient(server, token)
+        
+    def get_flow_id(self, flow_name):
+        flows = self.rapid_pro.get_flows().all(retry_on_rate_exceed=True)
+        matching_flows = [f for f in flows if f.name == flow_name]
+
+        if len(matching_flows) == 0:
+            available_flow_names = [f.name for f in flows]
+            raise KeyError(f"Requested flow not found on RapidPro (Available flows: {', '.join(available_flow_names)})")
+        if len(matching_flows) > 1:
+            raise KeyError("Non-unique flow name")
+
+        return matching_flows[0].uuid
+
+    def get_flow_ids(self, flow_names):
+        return [self.get_flow_id(name) for name in flow_names]
+
+    def get_flow_definitions_for_flow_ids(self, flow_ids):
+        return self.rapid_pro.get_definitions(flows=flow_ids, dependencies="all")
+
+    def get_raw_runs_for_flow_id(self, flow_id, range_start_inclusive=None, range_end_exclusive=None):
+        range_end_inclusive = None
+        if range_end_exclusive is not None:
+            range_end_inclusive = range_end_exclusive - datetime.timedelta(microseconds=1)
+
+        print(f"Fetching raw runs for flow with id '{flow_id}'...")
+        raw_runs = self.rapid_pro.get_runs(
+            flow=flow_id, after=range_start_inclusive, before=range_end_inclusive).all(retry_on_rate_exceed=True)
+        print(f"Fetched {len(raw_runs)} runs")
+
+        # Sort in ascending order of modification date
+        raw_runs = list(raw_runs)
+        raw_runs.reverse()
+
+        return raw_runs
+
+    def get_raw_contacts(self):
+        print("Fetching all raw contacts...")
+        raw_contacts = self.rapid_pro.get_contacts().all(retry_on_rate_exceed=True)
+        assert len(set(c.uuid for c in raw_contacts)) == len(raw_contacts), "Non-unique contact UUID in RapidPro"
+        print(f"Fetched {len(raw_contacts)} contacts")
+        return raw_contacts
+
+    def get_traced_runs_for_flow_id(self, user, flow_id, phone_uuids,
```

Comment:
I'd quite like a doc comment saying what phone_uuids should be - mainly because of its sensitivity

Reply:
already added in a future PR